src/hyperframe/frame.py

```diff
@@ -68,25 +68,52 @@ def __init__(self, stream_id, flags=()):
 
         if (not self.stream_id and
            self.stream_association == _STREAM_ASSOC_HAS_STREAM):
-            raise InvalidDataError('Stream ID must be non-zero')
+            raise InvalidDataError(
+                'Stream ID must be non-zero for {}'.format(
+                    type(self).__name__,
+                )
+            )
         if (self.stream_id and
            self.stream_association == _STREAM_ASSOC_NO_STREAM):
-            raise InvalidDataError('Stream ID must be zero')
+            raise InvalidDataError(
+                'Stream ID must be zero for {} with stream_id: {}'.format(
+                    type(self).__name__,
+                    self.stream_id,
+                )
+            )
 
     def __repr__(self):
-        flags = ", ".join(self.flags) or "None"
-        body = binascii.hexlify(self.serialize_body()).decode('ascii')
-        if len(body) > 20:
-            body = body[:20] + "..."
         return (
-            "{type}(Stream: {stream}; Flags: {flags}): {body}"
+            "{}(stream_id: {}; flags: {}): {}"
```

Comment:
Based on https://docs.python.org/3/library/functions.html#repr, I would make this so that repr() is proper Python code: 
```suggestion
            "{}(stream_id={}, {}, flags={})"
```

Reply:
👍 
added in 58d423d